probe/endpoint/reporter.go

```diff
@@ -71,6 +74,11 @@ func (r *Reporter) Stop() {
 	r.scanner.Stop()
 }
 
+type fourTuple struct {
+	fromAddr, toAddr string
+	fromPort, toPort uint16
+}
```

Comment:
Can you add a `func (f fourTuple) key() string {...}` function which does the stringing and sorting for the map key?  Ideally I think the seenTuples map should be a map[fourTuple]struct{} with a custom hash and equal on the fourTuples, but theres no operator overloading in go.